cpp/dolfinx/fem/DofMap.cpp

```diff
@@ -154,10 +161,11 @@ fem::transpose_dofmap(const graph::AdjacencyList<std::int32_t>& dofmap,
   std::vector<std::int32_t> data(index_offsets.back());
   std::vector<int> pos = index_offsets;
   int cell_offset = 0;
-  for (int c = 0; c < num_cells; ++c)
+  for (std::int32_t c = 0; c < num_cells; ++c)
   {
-    for (auto dof : dofmap.links(c))
-      data[pos[dof]++] = cell_offset++;
+    auto dofs = stdex::submdspan(dofmap, c, stdex::full_extent);
+    for (std::size_t d = 0; d < dofmap.extent(1); ++d)
+      data[pos[dofs[d]]++] = cell_offset++;
```

Comment:
Similar comment as above, as we never use anything but `dofs[d]` (never `d` itself).

Reply:
`mdspan` doesn't have `begin`/`end` (because it could be strided).